build.js

```diff
@@ -194,3 +196,29 @@ function prependToTestName(prefix) {
     };
 }
 
+/**
+ * Verifies that no tests have duplicate names.
+ * @param allTests - The list of all tests.
+ */
+function checkDuplicateNames(allTests) {
+    const names = new Set();
+    const duplicates = [];
+
+    for (const test of allTests) {
+        const oldSize = names.size;
+        const name = test.name;
+        names.add(name);
+
+        // If size remained the same, a duplicate was encountered
+        if (oldSize === names.size) {
```

Comment:
You can get rid of `oldSize` and do `if (names.has(test.name)) { ... }` instead.

Reply:
I was trying to make this a bit more efficient (hoping that `size` is not recalculated every time), because `has` + `add` would mean there are two 'contains' checks internally. It is quite unfortunate that `add` does not return whether it was successful.

But maybe that was premature optimization.

Reply:
I believe that however inefficient it is complexity-wise, it'll still be negligible compared to the IO.

Reply:
(I also tend to care less about efficiency on a build server.  Just my opinion, though.)